source/core/lethegridtools.cc

```diff
@@ -0,0 +1,982 @@
+//
+// Created by lucka on 2021-07-15.
+//
+
+#include <core/lethegridtools.h>
+#include <deal.II/grid/manifold.h>
+#include <deal.II/grid/manifold_lib.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/fe/fe_q.h>
+#include <cmath>
+
+
+template <int dim>
+void
+LetheGridTools::vertices_cell_mapping(const DoFHandler<dim> &dof_handler,
+        std::map<unsigned int,std::set<typename DoFHandler<dim>::active_cell_iterator>> & vertices_cell_map)
+{
+    // Find all the cells around each vertices
+
+    vertices_cell_map.clear();
+    const auto &cell_iterator = dof_handler.active_cell_iterators();
+
+
+    // // Loop on all the cells and find their vertices to fill the map of sets of
+    // cells around each vertex
+    for (const auto &cell : cell_iterator)
+    {
+        if (cell->is_locally_owned() || cell->is_ghost())
+        {
+            const unsigned int vertices_per_cell =
+                    GeometryInfo<dim>::vertices_per_cell;
+            for (unsigned int i = 0; i < vertices_per_cell; i++)
+            {
+                // First obtain vertex index
+                unsigned int v_index = cell->vertex_index(i);
+
+                // Insert the cell into the set of cell around that vertex.
+                vertices_cell_map[v_index].insert(cell);
+            }
+        }
+    }
+}
+
+
+template <int dim>
+typename DoFHandler<dim>::active_cell_iterator
+LetheGridTools::find_cell_around_point_with_tree(const DoFHandler<dim> &dof_handler,
+                                 Point<dim>             &point)
+{
+    // find cell around point using tree properties instead of looping on all cell
+
+    // define stuff for the search
+    MappingQ1<dim> map;
```

Comment:
```suggestion
    MappingQ1<dim> mapping;
```
don't call it a map, otherwise it becomes confusing with the std::map container.